### PR TITLE
Fix images in changes, problems and tickets notifications; fixes #4345

### LINKED ISSUE #4345: Broken Image in solution notification
No issue with followup and new ticket description.

1. Steps to reproduce (which actions have you made) :
- Apply bugfix: https://github.com/glpi-project/glpi/pull/4272 to solve "N/A" in solution notification.
- Add a picture inside a solution
image

2. Expected result :
- Picture is displayed in the mail

3. Actual result :
- A broken tag is displayed: impossible to show picture
image

4. Your GLPI setup (you can find it in Setup > General menu, System tab) :
9.3.0

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #4345 

Inline images were not correctly handled in solution, and pasting was not working in most cases.
I added the file field to enable JS handling of image pasting and added the call to `addFiles` method which replaces inline images by the stored image.

### CODE REVIEW

**Comment by cedric-anne** [comment]
I added a commit, but I am really not sure about it.

I use "mailhog" to catch mails on my dev machine. It does not interpret embedded images, so I use the "release ail" function to send it to my mailbox, and I read it in Zimbra web client.
When src is not the first attribute of the img tag, the mail I received show images as attachements, not as embedded.
When I look at the source code, I see some `;` chars on `img` tag string. Not sure they should be here.

**Comment by cedric-anne** [comment]
It's now fixed. Seems like that was an encoding problem. Mainly caused by https://github.com/glpi-project/glpi/pull/4401/files#diff-4ab777b773d4ae9eeccdbc10ca255a3cR276.

I added some changes to normalize line breaks and refactored small pieces of code.

I need to now apply HTML fields protection to changes and problems as well, and test it.

@trasher Does it looks good to you ?

File: inc/notificationtargetcommonitilobject.class.php [hunk 1]
```diff
@@ -1010,9 +1010,7 @@ function getDataForObject(CommonDBTM $item, array $options, $simple = false) {
             );
          }
 
-         $data["##$objettype.solution.description##"] = Toolbox::unclean_cross_side_scripting_deep(
-            $itilsolution->getField('content')
-         );
+         $data["##$objettype.solution.description##"] = $itilsolution->getField('content');
```
Comment (trasher) [inline inc/notificationtargetcommonitilobject.class.php]: A little doubt about this one

Comment (cedric-anne) [reply inc/notificationtargetcommonitilobject.class.php]: This is useless as it is done by the `Html::entity_decode_deep($data);` call on https://github.com/glpi-project/glpi/pull/4401/files#diff-4ab777b773d4ae9eeccdbc10ca255a3cR275.

**Comment by cedric-anne** [comment]
I added a recursive function to fix HTML protection of fields contained in an array of items. That fixes display of tickets in changes and problems, and also display of linked tickets of a ticket.
I also fixes followup display.

**Pull request merged**